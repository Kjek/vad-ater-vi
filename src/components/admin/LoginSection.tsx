import { signIn } from 'next-auth/react';
import { useRef } from 'react';
import InputButton from './Button';
import TextBox from './TextBox';

const LoginSection = () => {
  const userName = useRef('');
  const pass = useRef('');
  const onSubmit = async () => {
    const result = await signIn('credentials', {
      username: userName.current,
      password: pass.current,
      redirect: true,
      callbackUrl: '/admin',
    });
    console.log(result);
  };
  const keyDownEvent = (event: React.KeyboardEvent<HTMLDivElement>) => {
    if (event.code === 'Enter') {
      void onSubmit();
    }
  };
  return (
    <div
      className={
        'flex h-screen flex-col items-center justify-center gap-1 bg-neutral-100 dark:bg-gray-900'
      }
    >
      <div
        className='dark:bg-gray-custom flex flex-col gap-2 rounded-md bg-white px-7 py-4 shadow'
        onKeyDown={keyDownEvent}
      >
        <TextBox
          title='Username'
          placeholder='Username'
          onChange={(e) => (userName.current = e.target.value)}
        />
        <TextBox
          title='Password'
          placeholder='Password'
          variant='password'
          onChange={(e) => (pass.current = e.target.value)}
        />
        <InputButton value='Login' onClick={() => void onSubmit()} />
      </div>
    </div>
  );
};

export default LoginSection;
