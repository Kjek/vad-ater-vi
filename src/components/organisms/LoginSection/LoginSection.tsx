import InputButton from '@component/atoms/Button/Button';
import CustomForm from '@component/atoms/CustomForm/CustomForm';
import TextBox from '@component/atoms/TextBox/TextBox';
import CreateAdminAccountModal from '@component/molecules/CreateAdminAccountModal/CreateAdminAccountModal';
import { signIn } from 'next-auth/react';
import { useRef } from 'react';
import { useToggle } from 'usehooks-ts';

const LoginSection = () => {
  const [isGenAdminAccOpen, toggleGenAdminAccOpen] = useToggle();
  const userName = useRef('');
  const pass = useRef('');
  const onSubmit = async () => {
    await signIn('credentials', {
      username: userName.current,
      password: pass.current,
      redirect: true,
      callbackUrl: '/admin',
    });
  };

  return (
    <div
      className={
        'flex h-screen flex-col items-center justify-center gap-1 bg-neutral-100 dark:bg-gray-900'
      }
    >
      <CustomForm onSubmit={onSubmit}>
        <div className='dark:bg-gray-custom flex flex-col gap-2 rounded-md bg-white px-7 py-4 shadow'>
          <TextBox
            title='Username'
            placeholder='Username'
            onChange={(e) => (userName.current = e.target.value)}
            required
          />
          <TextBox
            title='Password'
            placeholder='Password'
            variant='password'
            onChange={(e) => (pass.current = e.target.value)}
            required
          />
          <InputButton type='submit' value='Login' />
        </div>
      </CustomForm>
      <div className='dark:bg-gray-custom absolute bottom-0 right-0 mb-4 mr-4 bg-white'>
        <InputButton
          value='Generate admin account'
          onClick={toggleGenAdminAccOpen}
        />
      </div>
      {isGenAdminAccOpen ? (
        <CreateAdminAccountModal toggleModal={toggleGenAdminAccOpen} />
      ) : null}
    </div>
  );
};

export default LoginSection;
