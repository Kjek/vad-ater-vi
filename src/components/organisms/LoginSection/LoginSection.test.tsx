import { render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import LoginSection from './LoginSection';

Date.prototype.getWeek = vi.fn();

describe(LoginSection, () => {
  it('renders correctly', () => {
    const { container } = render(<LoginSection />);
    expect(container).toBeInTheDocument();
    expect(container).toBeVisible();
  });

  it('matches snapshot', () => {
    const defaultRender = render(<LoginSection />);
    expect(defaultRender).toMatchSnapshot();
  });
});
