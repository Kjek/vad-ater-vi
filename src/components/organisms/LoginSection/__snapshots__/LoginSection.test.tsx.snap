// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginSection > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex h-screen flex-col items-center justify-center gap-1 bg-neutral-100 dark:bg-gray-900"
      >
        <div
          class="dark:bg-gray-custom flex flex-col gap-2 rounded-md bg-white px-7 py-4 shadow"
        >
          <input
            class="w-full self-center rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-800 outline-none focus:border-gray-300 focus:ring-gray-300 dark:border-gray-600 dark:bg-gray-800 dark:text-gray-300 dark:placeholder-gray-400 dark:focus:border-gray-600 dark:focus:ring-gray-600"
            placeholder="Username"
            title="Username"
            type="text"
          />
          <input
            class="w-full self-center rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-800 outline-none focus:border-gray-300 focus:ring-gray-300 dark:border-gray-600 dark:bg-gray-800 dark:text-gray-300 dark:placeholder-gray-400 dark:focus:border-gray-600 dark:focus:ring-gray-600"
            placeholder="Password"
            title="Password"
            type="password"
          />
          <input
            class="cursor-pointer rounded-lg border border-gray-300 px-5 py-2.5 text-sm font-medium text-gray-800 hover:bg-gray-200 dark:border-gray-600 dark:text-gray-300 dark:hover:bg-gray-700 dark:hover:text-white"
            type="button"
            value="Login"
          />
        </div>
        <div
          class="dark:bg-gray-custom absolute bottom-0 right-0 mb-4 mr-4 bg-white"
        >
          <input
            class="cursor-pointer rounded-lg border border-gray-300 px-5 py-2.5 text-sm font-medium text-gray-800 hover:bg-gray-200 dark:border-gray-600 dark:text-gray-300 dark:hover:bg-gray-700 dark:hover:text-white"
            type="button"
            value="Generate admin account"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex h-screen flex-col items-center justify-center gap-1 bg-neutral-100 dark:bg-gray-900"
    >
      <div
        class="dark:bg-gray-custom flex flex-col gap-2 rounded-md bg-white px-7 py-4 shadow"
      >
        <input
          class="w-full self-center rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-800 outline-none focus:border-gray-300 focus:ring-gray-300 dark:border-gray-600 dark:bg-gray-800 dark:text-gray-300 dark:placeholder-gray-400 dark:focus:border-gray-600 dark:focus:ring-gray-600"
          placeholder="Username"
          title="Username"
          type="text"
        />
        <input
          class="w-full self-center rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-800 outline-none focus:border-gray-300 focus:ring-gray-300 dark:border-gray-600 dark:bg-gray-800 dark:text-gray-300 dark:placeholder-gray-400 dark:focus:border-gray-600 dark:focus:ring-gray-600"
          placeholder="Password"
          title="Password"
          type="password"
        />
        <input
          class="cursor-pointer rounded-lg border border-gray-300 px-5 py-2.5 text-sm font-medium text-gray-800 hover:bg-gray-200 dark:border-gray-600 dark:text-gray-300 dark:hover:bg-gray-700 dark:hover:text-white"
          type="button"
          value="Login"
        />
      </div>
      <div
        class="dark:bg-gray-custom absolute bottom-0 right-0 mb-4 mr-4 bg-white"
      >
        <input
          class="cursor-pointer rounded-lg border border-gray-300 px-5 py-2.5 text-sm font-medium text-gray-800 hover:bg-gray-200 dark:border-gray-600 dark:text-gray-300 dark:hover:bg-gray-700 dark:hover:text-white"
          type="button"
          value="Generate admin account"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
